students/lee_deitesfeld/lesson04/mailroom.py

```diff
@@ -0,0 +1,139 @@
+        x['sum'] = ttl_amt
+
+    #calculates number of donations and adds new key-value pair to donor_db dict
+    for x in dic_donors['donor_db']:
+        numofdonations = len(x['donations'])
+        x['count'] = numofdonations
+
+    #calculates average of donations and adds new key-value pair to donor_db dict
+    for x in dic_donors['donor_db']:
+        average = sum(x['donations'])/len(x['donations'])
+        str_average = f'{average:.2f}' #limits average to two decimal points
+        x['average'] = str_average
+
+    #sorts list of donors on total amount donated
+    for x in dic_donors['donor_db']:
+        int_ttl_amt = int(x['sum']) #convert ttlamt to integer so sort will work on entire digit (not just first)
+        x['sum'] = int_ttl_amt
+    dic_donors['donor_db'].sort(key=itemgetter('sum'), reverse = True) #sort on key='sum'
+    for x in dic_donors['donor_db']:
+        ttl_amt_2dec = '%.2f' % x['sum'] #limits total to two decimal points
+        x['sum'] = ttl_amt_2dec
+
+    #removes history of donations, since it's not in final report
+    for x in dic_donors['donor_db']:
+        del x['donations']
+
+    #formats final report
+    header = ["Name", "Donations Total", "Number of Donations", "Average Donation"]
+    print('{:>10} {:>20} {:>30} {:>40}'.format(*header))
+    print('---'*35)
+    for line in dic_donors['donor_db']:
+        print('{:>10} {:>20} {:>30} {:>40}'.format(*line.values()))
+
+def listof_donors():
+    '''Corresponds to submenu item List the Names of Donors'''
+    for item in dic_donors['donor_db']:
+        print(item['name'])
+
+def add_donor():
+    '''Corresponds to submenu item Add a New Donation'''
+    donor_question = input("Please enter the name of a new or existing donor: ")
+    for item in dic_donors['donor_db']:
+        if donor_question in item["name"]:
+            donation_amount = input("Please enter donation amount for donor: ")
+            item["latest_donation"] = donation_amount #adds temporary key-value pair for email printout
+            email = 'Dear {name}, Thank you for your generous donation of {latest_donation}. We hope you will consider donating again!'.format(**item)
+            print(email)
+            del item["latest_donation"] #deletes temporary key-value pair
+            item['donations'].append(float(donation_amount)) #adds donation to current list of donations
+        else:
+            continue
+    if donor_question not in item["name"]:
+        dic_new_donor= {}
+        dic_new_donor["name"] = donor_question
+        dic_donors['donor_db'].append(dic_new_donor)
+        donation_amount = input("Please enter donation amount for donor: ")
+        dic_new_donor['donations'] = donation_amount
+        email = 'Dear {name}, Thank you for your generous donation of {donations}. We hope you will consider donating again!'.format(**dic_new_donor)
+        print(email)
+        dic_new_donor['donations'] = [float(donation_amount)]
```

Comment:
switch your main data structure to a dictionary and the code in this function can be reduced in half.